### PR TITLE
Handling student's added libraries in makefile

### PR DESCRIPTION
**Pull request by QLaille**
Following #360 , I've modified every makefile of every track to rely on a LIBS variable referencing extra library flags for an exercise, commented how to add a library and added to the compilation line said variable. (I also still suck a bit with git and it shows previous commits).

### CODE REVIEW

File: exercises/secret-handshake/makefile [hunk 1]
```diff
@@ -1,3 +1,7 @@
+### If you wish to use extra libraries (math.h for instance),
+### add their flags here (-lm in our case) in the "LIBS" variable.
```
Comment (ryanplusplus) [inline exercises/secret-handshake/makefile]: My only nitpick is that we include `(-lm in our case)` even when we don't have `-lm` (or any `LIBS` variable at all).

**Comment by QLaille** [comment]
Silly me doing silly things with the clipboard, hang on.

**Comment by ryanplusplus** [comment]
Looks good to me.

@patricksjackson any thoughts before merge?

**Comment by patricksjackson** [comment]
I think it's fine, I'll go ahead and merge. I also was unaware that some of our Makefiles were customized for the exercise. I'll try to remember that in future.

Thanks @QLaille !

Closes #360

**Pull request merged**